src/display_array.jl

```diff
@@ -0,0 +1,55 @@
+export DisplayArray, render
+
+struct DisplayArray
+    size::Tuple{Int, Int}
+    background::Char
+    context::Vector{DisplayRow}
+end
+
+function DisplayArray(h, w; background=' ')
+    (textwidth(background) != 1) && (throw("Bad background char"))
+
+    context = DisplayRow[]
+    for i=1:h
+        push!(context, DisplayRow(w, background=background))
+    end
+
+    return DisplayArray(
+        (h, w),
+        background,
+        context
+    )
+end
+
+Base.size(array::DisplayArray) = array.size
+
+Base.lastindex(array::DisplayArray, d::Int) = size(array)[d]
+
+function Base.setindex!(
+    array::DisplayArray,
+    c::Char,
+    display_row::Int, display_col::Int
+)
+    array.context[display_row][display_col] = c
+end
+
+function Base.setindex!(
+    array::DisplayArray,
+    str::String,
+    display_row_range::UnitRange{Int64}, display_col_range::UnitRange{Int64}
+)
+    split_str = split(str, '\n')
+    start, stop = display_row_range.start, display_row_range.stop
+    (stop != start + length(split_str) - 1) && (throw(DimensionMismatch))
```

Comment:
Is this what you want?
```suggestion
  length(split_str) == length(display_row_range) || throw(DimensionMismatch)
```